### PR TITLE
Fix posixtest

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Fixed kernel:posixtest
1-fixed pep8 issue
2-removed unnecessary patch
3.a- added ignore_status=True in process.system to avoid if command fail script need not to
error out
 b. added test case check if test fail , test need to be fail
4-Added all test option can be run as via yaml param
added yaml file as well

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
````conformance/interfaces/pthread_spin_init/1-1: link: FAILED 
conformance/interfaces/pthread_spin_init/2-1: link: FAILED 
conformance/interfaces/pthread_spin_init/2-2: link: FAILED 
conformance/interfaces/pthread_spin_init/4-1: link: FAILED 
conformance/interfaces/pthread_spin_init/1-1: link: FAILED 
conformance/interfaces/pthread_spin_init/2-1: link: FAILED 
conformance/interfaces/pthread_spin_lock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_lock/3-2: link: FAILED 
conformance/interfaces/pthread_spin_lock/1-2: link: FAILED 
conformance/interfaces/pthread_spin_lock/3-1: link: FAILED 
conformance/interfaces/pthread_spin_lock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_lock/3-2: link: FAILED 
conformance/interfaces/pthread_spin_lock/1-2: link: FAILED 
conformance/interfaces/pthread_spin_lock/3-1: link: FAILED 
conformance/interfaces/pthread_spin_trylock/4-1: link: FAILED 
conformance/interfaces/pthread_spin_trylock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_trylock/4-1: link: FAILED 
conformance/interfaces/pthread_spin_trylock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_unlock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_unlock/1-2: link: FAILED 
conformance/interfaces/pthread_spin_unlock/3-1: link: FAILED 
conformance/interfaces/pthread_spin_unlock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_unlock/1-2: link: FAILED 
conformance/interfaces/pthread_spin_unlock/3-1: link: FAILED 
conformance/interfaces/pthread_testcancel/1-1: link: FAILED 
conformance/interfaces/pthread_testcancel/2-1: link: FAILED 
conformance/interfaces/pthread_testcancel/1-1: link: FAILED 
conformance/interfaces/pthread_testcancel/2-1: link: FAILED 
 -> self <class 'posixtest.Posixtest'>: 1-posixtest.py:Posixtest.test
 -> logfile <type 'str'>: /root/arpana/avocado-fvt-wrapper/results/job-2018-05-11T00.35-eb910d5/test-results/1-posixtest.py_Posixtest.test/stdout
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
FAIL 1-posixtest.py:Posixtest.test -> TestFail: test failed, Please check debug log for failed test cases

Test results available in /root/arpana/avocado-fvt-wrapper/results/job-2018-05-11T00.35-eb910d5```

File: kernel/posixtest.py [hunk 1]
```diff
@@ -60,7 +60,14 @@ def setUp(self):
     def test(self):
 
         os.chdir(self.sourcedir)
-        process.system('./run_tests THR')
+        process.system('./run_tests THR', ignore_status=True)
+
+        logfile = os.path.join(self.logdir, "stdout")
+        failed_tests = process.system_output(
```
Comment (narasimhan-v) [inline kernel/posixtest.py]: It is better to do this through parsing the output via `process.system_output('./run_tests THR')` or native file handling of stdout.

Comment (PraveenPenguin) [reply kernel/posixtest.py]: @narasimhan-v thanks for review , sure make sense will adopt the changes

**Comment by PraveenPenguin** [comment]
```conformance/interfaces/pthread_spin_unlock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_unlock/1-2: link: FAILED 
conformance/interfaces/pthread_spin_unlock/3-1: link: FAILED 
conformance/interfaces/pthread_spin_unlock/1-1: link: FAILED 
conformance/interfaces/pthread_spin_unlock/1-2: link: FAILED 
conformance/interfaces/pthread_spin_unlock/2-1: execution: PASS
conformance/interfaces/pthread_spin_unlock/3-1: link: FAILED 
conformance/interfaces/pthread_testcancel/1-1: link: FAILED 
conformance/interfaces/pthread_testcancel/2-1: link: FAILED 
conformance/interfaces/pthread_testcancel/1-1: link: FAILED 
conformance/interfaces/pthread_testcancel/2-1: link: FAILED 
****Tests Complete****
 -> error <type 'bool'>: True
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
FAIL 1-posixtest.py:Posixtest.test -> TestFail: Testcase failed during test check the log

Test results available in /root/arpana/avocado-fvt-wrapper/results/job-2018-05-16T07.15-0169b2e```

**Comment by PraveenPenguin** [comment]
@narasimhan-v done with changes can you please have a look

**Pull request merged**